src/odb/src/lefin/lefin.cpp

```diff
@@ -899,18 +899,12 @@ void lefin::layer(lefiLayer* layer)
                               : l->createDefaultAntennaRule();
       cur_model = layer->antennaModel(j);
       if (cur_model->hasAntennaAreaFactor()) {
-        if (cur_model->hasAntennaAreaFactorDUO())
-          cur_ant_rule->setAreaFactor(-1.0, cur_model->antennaAreaFactor());
-        else
-          cur_ant_rule->setAreaFactor(cur_model->antennaAreaFactor());
+          cur_ant_rule->setAreaFactor(cur_model->antennaAreaFactor(), cur_model->hasAntennaAreaFactorDUO());
```

Comment:
For context, hasAntennaAreaFactorDUO() returns true if the DiffuseOnly flag is used in the AntennaAreaFactor property.

The setAreaFactor() function takes the factor and the diffuseOnly flags as parameters. 
```cpp
void setAreaFactor(double factor, bool diffuse = false)
```
It is clear that the arguments passed to the function call in line 903 do not really match the expected input. In the old implementation, we pass -1 for the ```factor``` and the actual value of the factor in place of the ```diffuse``` flag.

Also, passing -1 as the factor to the setAreaFactor() function causes an assertion to fail because the function assertsthat the factor is positive. Consequently, when hasAntennaAreaFactorDUO() returns true, the assertion fails and execution stops.

To fix these issues, it made more sense to pass:
1- the area factor as the first argument
2- the returned value from hasAntennaAreaFactorDUO() as the second argument.